apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexBeanProducer.java

```diff
@@ -0,0 +1,56 @@
+package com.apollocurrency.aplwallet.apl.exchange.service;
+
+import com.apollocurrency.aplwallet.apl.util.StringUtils;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.http.HttpService;
+import org.web3j.tx.response.PollingTransactionReceiptProcessor;
+import org.web3j.tx.response.TransactionReceiptProcessor;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import static org.web3j.protocol.core.JsonRpc2_0Web3j.DEFAULT_BLOCK_TIME;
+import static org.web3j.tx.TransactionManager.DEFAULT_POLLING_ATTEMPTS_PER_TX_HASH;
+
+@Singleton
+public class DexBeanProducer {
+
+    private PropertiesHolder propertiesHolder;
+    private static Web3j web3j;
+    private static TransactionReceiptProcessor transactionReceiptProcessor;
+
```

Comment:
What about double lock pattern (see SafeLocalDCLFactory in https://shipilev.net/blog/2014/safe-public-construction/) ?